server/service/orbit.go

```diff
@@ -450,3 +451,83 @@ func (svc *Service) SaveHostScriptResult(ctx context.Context, result *fleet.Host
 
 	return fleet.ErrMissingLicense
 }
+
+/////////////////////////////////////////////////////////////////////////////////
+// Post Orbit disk encryption key
+/////////////////////////////////////////////////////////////////////////////////
+
+type orbitPostDiskEncryptionKeyRequest struct {
+	OrbitNodeKey  string `json:"orbit_node_key"`
+	EncryptionKey []byte `json:"encryption_key"`
+	ClientError   string `json:"client_error"`
+}
+
+// interface implementation required by the OrbitClient
+func (r *orbitPostDiskEncryptionKeyRequest) setOrbitNodeKey(nodeKey string) {
+	r.OrbitNodeKey = nodeKey
+}
+
+// interface implementation required by orbit authentication
+func (r *orbitPostDiskEncryptionKeyRequest) orbitHostNodeKey() string {
+	return r.OrbitNodeKey
+}
+
+type orbitPostDiskEncryptionKeyResponse struct {
+	Err error `json:"error,omitempty"`
+}
+
+func (r orbitPostDiskEncryptionKeyResponse) error() error { return r.Err }
+func (r orbitPostDiskEncryptionKeyResponse) Status() int  { return http.StatusNoContent }
+
+func postOrbitDiskEncryptionKeyEndpoint(ctx context.Context, request interface{}, svc fleet.Service) (errorer, error) {
+	req := request.(*orbitPostDiskEncryptionKeyRequest)
+	if err := svc.SetOrUpdateDiskEncryptionKey(ctx, string(req.EncryptionKey), req.ClientError); err != nil {
+		return orbitPostDiskEncryptionKeyResponse{Err: err}, nil
+	}
+	return orbitPostDiskEncryptionKeyResponse{}, nil
+}
+
+func (svc *Service) SetOrUpdateDiskEncryptionKey(ctx context.Context, encryptionKey, clientError string) error {
+	// this is not a user-authenticated endpoint
+	svc.authz.SkipAuthorization(ctx)
+
+	host, ok := hostctx.FromContext(ctx)
+	if !ok {
+		return newOsqueryError("internal error: missing host from request context")
+	}
+
+	if !svc.config.MDM.IsMicrosoftWSTEPSet() || !config.IsMDMFeatureFlagEnabled() {
```

Comment:
should we use the windows middleware instead? if that's too tricky because this is an orbit endpoint then should we check if the host is enrolled into windows MDM?

Reply:
Oh! Great catch, yeah I think the middleware can be added even if it's an orbit endpoint, let me update this. That being said, sounds like it should _also_ check that the host is enrolled into MDM?